src/main/java/com/hcl/appscan/jenkins/plugin/scanModes/FullScan.java

```diff
@@ -0,0 +1,109 @@
+package com.hcl.appscan.jenkins.plugin.scanModes;
+
+import hudson.Extension;
+import hudson.Util;
+import hudson.util.Secret;
+import hudson.util.VariableResolver;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+import java.util.Map;
+
+public class FullScan extends ScanMode {
+
+    private static final String FULL_SCAN = "Full Scan"; //$NON-NLS-1$
+
+    private String m_loginType;
+    private String m_trafficFile;
+    private String m_userName;
+    private Secret m_password;
+    private String m_exploreData;
+
+    @DataBoundConstructor
+    public FullScan(String loginType, String accessId, String secretKey, String trafficFile, String exploreData) {
+        m_loginType = loginType;
+        m_userName = accessId;
+        m_password = Secret.fromString(secretKey);
+        m_trafficFile = trafficFile;
+        m_exploreData = exploreData;
+    }
+
+    @DataBoundSetter
+    public void setLoginType(String loginType) {
+        m_loginType = loginType;
+    }
+
+    public String getLoginType() {
+        return m_loginType;
+    }
+
+    @DataBoundSetter
+    public void setAccessId(String userName) {
+            m_userName = userName;
+    }
+
+    public String getAccessId() {
+        return m_userName;
+    }
+
+    @DataBoundSetter
+    public void setSecretKey(String password) {
+            m_password = Secret.fromString(password);
+    }
+
+    public String getSecretKey() {
+        return Secret.toString(m_password);
+    }
+
+    @DataBoundSetter
+    public void setTrafficFile(String trafficFile) {
+            m_trafficFile = trafficFile;
+    }
+
+    public String getTrafficFile() {
+        return m_trafficFile;
+    }
+
+    @DataBoundSetter
+    public void setExploreData(String exploreData) {
+        m_exploreData = exploreData;
+    }
+
+    public String getExploreData() {
+        return m_exploreData;
+    }
+
+    public String isLoginType(String loginTypeName) {
```

Comment:
Is there a reason isLoginType returns a String instead of a boolean? If it needs to be a String the logic could be cleaned up to just be:
`return loginTypeName.equalsIgnoreCase(m_loginType) || loginTypeName.equals("Manual") ? "true" : "";`

Reply:
Yes, LoginType needs to be a String. Updating the methods in both classes as per the suggestion.